core/src/main/scala/pink/cozydev/lucille/Query.scala

```diff
@@ -50,22 +56,81 @@ object MultiQuery {
 }
 
 object Query {
+
+  /** A term query
+    * e.g. 'cat', 'catch22'
+    *
+    *  @param str the term
+    */
   final case class Term(str: String) extends TermQuery {
     override def mapLastTerm(f: Query.Term => Query): Query =
       f(this)
   }
+
+  /** A phrase query
+    * e.g. 'the cat jumped'
+    *
+    * @param str the phrase
+    */
   final case class Phrase(str: String) extends TermQuery
+
+  /** A prefix query
+    * Search for words starting with the given prefix
+    * e.g. 'jump*'
+    *
+    * @param str the prefix
+    */
   final case class Prefix(str: String) extends TermQuery
+
+  /** A proximity query
+    * Search for words within a specified word distance
+    * e.g. '"cat jumped"~3', '"one two three"~2'
+    *
+    * @param str the words
+    * @param num the word distance
+    */
   final case class Proximity(str: String, num: Int) extends TermQuery
+
+  /** A fuzzy query
+    * Search for words within a Damerau-Levenshtein Distance
+    * The additional parameter, between 0 and 2, specifies the number of edits allowed (defaults to 2)
```

Comment:
```suggestion
  /** A fuzzy query with an optional distance value
```

> The additional parameter, between 0 and 2, specifies the number of edits allowed (defaults to 2)

I think these are semantics(?) from Lucene that we don't necessarily need to copy over.